modopt/tests/test_opt.py

```diff
@@ -723,349 +286,229 @@ def setUp(self):
-            err_msg='Incorrect low rank operation: standard with rank',
-        )
-        npt.assert_almost_equal(
-            self.lowrank_ngole.op(self.data3),
-            self.data5,
-            err_msg='Incorrect low rank operation: ngole',
-        )
-
-        npt.assert_almost_equal(
-            self.lowrank.cost(self.data3, verbose=True),
-            469.39132942464983,
-            err_msg='Incorrect low rank cost.',
-        )
-
-    def test_linear_comp_prox(self):
-        """Test LinearCompositionProx."""
-        npt.assert_array_equal(
-            self.linear_comp.op(self.data1),
-            self.data2,
-            err_msg='Incorrect sparse threshold operation.',
-        )
-
-        npt.assert_equal(
-            self.linear_comp.cost(self.data1, verbose=True),
-            108.0,
-            err_msg='Incorrect sparse threshold cost.',
+    def case_prox_identity(self):
+        """Case prox identity."""
+        return proximity.IdentityProx(), 3, 3, 0
+
+    def case_prox_positivity(self):
+        """Case prox positivity."""
+        return proximity.Positivity(), -3, 0, 0
+
+    def case_prox_sparsethresh(self):
+        """Case prox sparsethreshosld."""
+        return (
+            proximity.SparseThreshold(linear.Identity(), weights=self.weights),
+            self.array33,
+            self.array33_st,
+            108,
+        )
+
+    @parametrize(
+        "lowr_type, initial_rank, operator, result, cost",
+        [
+            ("standard", None, None, array233_2, 469.3913294246498),
+            ("standard", 1, None, array233_2, 469.3913294246498),
+            ("ngole", None, func_double, array233_3, 469.3913294246498),
+        ],
+    )
+    def case_prox_lowrank(self, lowr_type, initial_rank, operator, result, cost):
+        """Case prox lowrank."""
+        return (
+            proximity.LowRankMatrix(
+                10,
+                lowr_type=lowr_type,
+                initial_rank=initial_rank,
+                operator=operator,
+                thresh_type="hard" if lowr_type == "standard" else "soft",
+            ),
+            self.array233,
+            result,
+            cost,
         )
 
-    def test_proximity_combo(self):
-        """Test ProximityCombo."""
-        for data7, data8 in zip(self.data7, self.data8):
-            npt.assert_array_equal(
-                data7,
-                data8,
-                err_msg='Incorrect combined operation',
+    def case_prox_linear_comp(self):
+        """Case prox linear comp."""
+        return (
+            proximity.LinearCompositionProx(
+                linear_op=linear.Identity(), prox_op=self.case_prox_sparsethresh()[0]
+            ),
+            self.array33,
+            self.array33_st,
+            108,
+        )
+
+    def case_prox_ridge(self):
+        """Case prox ridge."""
+        return (
+            proximity.Ridge(linear.Identity(), self.weights),
+            self.array33 * (1 + 1j),
```

Comment:
Why do you perform this test on complex-valued data? Weird ...

Reply:
I simply adapted the existing test.

Reply:
I know but this does not really make sense to start with POCS